src/main/kotlin/step2/calculator/Calculator.kt

```diff
@@ -0,0 +1,34 @@
+package step2.calculator
+
+import step2.calculator.StringExpression.parseToMathExpression
+
+object Calculator {
+    private const val ARITHMETIC_EXCEPTION_ERROR_MESSAGE = "Error occurred during calculate"
+
+    fun calculate(expression: String): Int {
+        val mathExpression = parseToMathExpression(expression)
+
+        return kotlin.runCatching {
+            calculate(mathExpression)
+        }.onFailure {
+            throw ArithmeticException(ARITHMETIC_EXCEPTION_ERROR_MESSAGE)
+        }.getOrThrow()
+    }
+
+    private fun calculate(mathExpression: MathExpression): Int {
+        var result: Int = mathExpression.operands[0].value
+
+        for (i in 0 until mathExpression.operatorsSize) {
+            val operator: Operator = mathExpression.operators[i]
+            val operand: Operand = mathExpression.operands[i + 1]
```

Comment:
```suggestion
            val operator = mathExpression.operators[i]
            val operand = mathExpression.operands[i + 1]
```
타입 추론이 가능한 경우에는 타입을 생략해도 좋을 것 같아요

Reply:
저 또한 타입 추론이 가능하다면 최대한 줄이는게 맞다고 생각됩니다만
많은 분들이 실수를 방지하기 위하여 함수 또는 변수에 타입을 작성한다는 얘기가 많아 이렇게 작성했던걸로 기억하는데
이에 대해서는 어떻게 생각하시나요??
보통 회사에서 사용시에 이런 부분까지 컨벤션으로 정해두시나요?
(제가 다니는 곳은 이런 부분까지 컨벤션으로 정해두지는 않았습니다)

Reply:
요즘에는 IDE가 너무 좋아서 hint 설정을 해두시면 어떤 타입인지 잘 보입니다
물론 타입을 매번 명시해주는 것이 좋지만, 코틀린에서 지원하는 타입 추론 기능을 활용하는 것이 좀 더 코틀린스럽게 사용하는 것 같아요

Reply:
네 제거 했습니다~